### DISCUSSION TITLE
TypeScript intellisense for server's built-in events

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [socketio/socket.io#4751] [Q&A]
I followed the guide on typing the events sent from server to client and vice-versa,
but I'm not getting completion of the built-in events of the server perse, like `connection`.

![image](https://github.com/socketio/socket.io/assets/10191085/04fd9405-6f1c-4c31-85ce-2bd31bdcd5bc)

1. Is this a missing feature? 
2. Is my only option to make my own interface for the Server and having an union type of all the server built-in events?
Thank you.

**Comment by darrachequesne** [comment]
That's weird, built-in events should be supported (added in v4.0.0):

![Screenshot from 2023-06-26 08-12-15](https://github.com/socketio/socket.io/assets/13031701/2fe4c9fb-2624-4a21-857a-affa57cc8d39)

Which IDE are you using?

**Reply by angelhdzmultimedia** [reply]
### Environment

VSCode: **1.79.2**
typescript: **5.1.3**
socket.io: **4.7.0**

### Screenshots

![image](https://github.com/socketio/socket.io/assets/10191085/d132962e-e43a-48da-b0d7-72cbbc050124)

![image](https://github.com/socketio/socket.io/assets/10191085/7ee90bc4-0825-4875-a416-5c0678b8661f)

![image](https://github.com/socketio/socket.io/assets/10191085/0e00cb83-e905-4d3c-bbf4-789008f4b2d5)


### Final Words
It's weird. I have been working with TypeScript in different frameworks (NestJS, Nuxt 3, Vue, Vanilla, Node/Express) and TypeScript intellisense has been working great. Of course I always restart TypeScript lang server, and even reload the workspace just in case.
